### PR TITLE
move_to!

### PR DESCRIPTION
**Pull request by ieric01**
-rails s command functional
-move_to! method allows current piece to move to destination point if there's no piece
-Rspec tests

### CODE REVIEW

File: config/boot.rb [hunk 1]
```diff
@@ -1,3 +1,13 @@
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
 require 'bundler/setup' # Set up gems listed in the Gemfile.
+require 'rails/commands/server'
+
+module Rails
+ class Server
+   alias :default_options_alias :default_options
+   def default_options
+     default_options_alias.merge!(:Host => '0.0.0.0')
```
Comment (mscoutermarsh) [inline config/boot.rb]: nice

File: app/models/piece.rb [hunk 2]
```diff
@@ -85,9 +85,14 @@ def horizontal_obstruction?(destination_x, _destination_y)
 
   def move_to!(destination_x, destination_y)
     piece = Piece.find_by(x: destination_x, y: destination_y)
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: I'd call this `destination_piece` to be more clear that it's different

File: app/models/piece.rb [hunk 3]
```diff
@@ -85,9 +85,14 @@ def horizontal_obstruction?(destination_x, _destination_y)
...
-    if piece.color != color
-      piece.destroy
-      update(x: destination_x, y: destination_y)
+
+    if piece.present?
+      if piece.color != color
+        piece.destroy
+        update(x: destination_x, y: destination_y)
+      end
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: What happens if the piece is the same color?

File: app/models/piece.rb [hunk 4]
```diff
@@ -85,9 +85,14 @@ def horizontal_obstruction?(destination_x, _destination_y)
...
+    else
+        update(x: destination_x, y: destination_y)
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: line 92 and 95 have duplication. Any way you can call that just once?

File: app/models/piece.rb (hunk 3 above)
Comment (ieric01) [inline app/models/piece.rb]: If the piece is the same color, then nothing happens the xy coordinates of the current piece will not be updated

File: app/models/piece.rb (hunk 2 above)
Comment (ieric01) [inline app/models/piece.rb]: will do
.

**Pull request merged**